ert_shared/ensemble_evaluator/prefect_ensemble/prefect_ensemble.py

```diff
@@ -0,0 +1,317 @@
+                    )
+                    result = stage_task(expected_res=inputs)
+
+                    for o in step.get("outputs", []):
+                        o_t_res[o] = result["outputs"]
+        return flow
+
+    def evaluate(self, config, ee_id):
+        self._ee_dispach_url = config.get("dispatch_url")
+        self._ee_id = ee_id
+        self._eval_proc = multiprocessing.Process(
+            target=self._evaluate,
+            args=(self._ee_dispach_url, ee_id),
+        )
+        self._eval_proc.daemon = True
+        self._eval_proc.start()
+
+    def _evaluate(self, dispatch_url, ee_id):
+        sys.stderr = open("prefect.log", "w")
+        coef_input_files = gen_coef(
+            self.config["parameters"], self.config["realizations"], self.config
+        )
+
+        real_per_batch = self.config["max_running"]
+        real_range = range(self.config["realizations"])
+
+        storage = storage_driver_factory(
+            self.config.get("storage"), self.config.get("config_path", ".")
+        )
+        for stage in self.config["stages"]:
+            for step in stage["steps"]:
+                res_list = []
+                for res in step["resources"]:
+                    res_list.append(storage.store(res))
+                step["resources"] = res_list
+
+        with Client(dispatch_url) as c:
+            event = CloudEvent(
+                {
+                    "type": ids.EVTYPE_ENSEMBLE_STARTED,
+                    "source": f"/ert/ee/{self._ee_id}",
+                    "datacontenttype": "application/json",
+                },
+            )
+            c.send(to_json(event).decode())
+
+        i = 0
+        while i < self.config["realizations"]:
+            realization_range = real_range[i : i + real_per_batch]
+            flow = self.get_flow(
+                ee_id, dispatch_url, coef_input_files, realization_range
+            )
+            flow.run(executor=_get_executor(self.config["executor"]))
+            i = i + real_per_batch
+
+        with Client(dispatch_url) as c:
+            event = CloudEvent(
+                {
+                    "type": ids.EVTYPE_ENSEMBLE_STOPPED,
+                    "source": f"/ert/ee/{self._ee_id}",
+                    "datacontenttype": "application/json",
+                },
+            )
+            c.send(to_json(event).decode())
+
+    def is_cancellable(self):
+        return True
+
+    def cancel(self):
+        threading.Thread(target=self._cancel).start()
+
+    def _cancel(self):
+        if self._eval_proc is not None:
+            os.kill(self._eval_proc.pid, signal.SIGINT)
+        self._eval_proc = None
+        event = CloudEvent(
+            {
+                "type": ids.EVTYPE_ENSEMBLE_CANCELLED,
+                "source": f"/ert/ee/{self._ee_id}",
+                "datacontenttype": "application/json",
+            },
+        )
+
+        loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(loop)
```

Comment:
Could this be removed, since we are running the loop explicitly?

Reply:
yes